lib/APPRemoteControl/App.cpp

```diff
@@ -155,16 +158,78 @@ void App::loop()
         m_sendLineSensorsDataInterval.restart();
     }
 
-    if (true == m_reportTimer.isTimeout())
+    m_smpServer.process(millis());
+
+    m_systemStateMachine.process();
+}
+
+void App::handleRemoteCommand(const Command& cmd)
+{
+    CommandResponse rsp = {cmd.commandId, SMPChannelPayload::RSP_ID_OK};
+
+    switch (cmd.commandId)
     {
-        /* Send current data to SerialMuxProt Client */
-        reportVehicleData();
+    case SMPChannelPayload::CmdId::CMD_ID_IDLE:
+        /* Nothing to do. */
+        break;
 
-        m_reportTimer.restart();
+    case SMPChannelPayload::CmdId::CMD_ID_START_LINE_SENSOR_CALIB:
+        m_systemStateMachine.setState(&LineSensorsCalibrationState::getInstance());
+        break;
+
+    case SMPChannelPayload::CmdId::CMD_ID_REINIT_BOARD:
+        /* Ensure that the motors are stopped, before re-initialize the board. */
+        DifferentialDrive::getInstance().setLinearSpeed(0, 0);
+
+        /* Re-initialize the board. This is required for the webots simulation in
+         * case the world is reset by a supervisor without restarting the RadonUlzer
+         * controller executable.
+         */
+        Board::getInstance().init();
+        break;
+
+    case SMPChannelPayload::CmdId::CMD_ID_GET_MAX_SPEED:
+        rsp.maxMotorSpeed = Board::getInstance().getSettings().getMaxSpeed();
+        break;
+
+    case SMPChannelPayload::CmdId::CMD_ID_SET_INIT_POS:
+        Odometry::getInstance().clearPosition();
+        Odometry::getInstance().clearMileage();
+        Odometry::getInstance().setPosition(cmd.xPos, cmd.yPos);
+        Odometry::getInstance().setOrientation(cmd.orientation);
+
+        StartupState::getInstance().notifyInitialDataIsSet();
+        break;
+
+    default:
+        /* Command not known/relevant to the application. */
+        rsp.responseId = SMPChannelPayload::RSP_ID_ERROR;
```

Comment:
not known/relevant is not the same as ERROR typically.
Consider a "UNKNOWN" return to help caller to handle ERROR and not implemented/relevant differently.